highfidelity/hifi
# Fix some Model loading issues with the translucent textures

Trying to address issues with the interpretation of the alpha channel of the albedo map.
Making sure we are honoring our Material Guide correctly:
http://hifi-public.s3.amazonaws.com/sam/MaterialExportGuide/Material%20Export%20Guide.html

- Cleaned up the names and bit flags for the Translucent / OpacityMask Map and TranslucentFactor.
- The albedoMap, if detected as Translucent texture is then preprocessed to detect the level of opacity:
 -- Opaque (no alpha or all = 1),
 -- OpacityMask (0 or 1)
 -- Translucent (true use of the full range)
- The material is categorized as:  "SurfaceOpaque / TexelOpaque / Translucent" based on the combination of the albedoMap translucency property and the translucencyFactor.

- REvised the shader code to have a more systematic way of getting the "opacity" information in the fragment shader.

TO TEST:
Try this file:
https://s3-us-west-1.amazonaws.com/hifi-content/DomainContent/Junkyard/Playa/transparentcube.fbx
It should be seen as transparent.
Using the FrameDebugger.js you could see it s indeed not rendered has opaque but instead as transparent

This file should not be rendered as transparent, it is the same model, same texture but not bound to "transparency" field of maya material:
https://s3-us-west-1.amazonaws.com/hifi-content/DomainContent/Junkyard/Playa/breaktransparentcube.fbx
In the master build this would be seen transparent

Simirlay with this 2 files testing the alpha channel used as OpacityMask:

This shoudl work in both builds and be seen in the debugFramebuffer.js as opaque
https://s3-us-west-1.amazonaws.com/hifi-content/DomainContent/Junkyard/Playa/leaf_transparent.fbx

This file should be fully opaque (grey box + the leaf texture) in this build
It should be broken in the current master build
https://s3-us-west-1.amazonaws.com/hifi-content/DomainContent/Junkyard/Playa/leaf_brokentransparency.fbx
